src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,27 +399,27 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
-// TO_NUMERIC (int or double)
-auto InvalidArg = std::errc::invalid_argument;
-auto OutOfRange = std::errc::result_out_of_range;
-
+// Expressions that convert a string to a number (int or double).
 template <typename T>
 [[maybe_used]] auto toNumeric = [] (std::optional<std::string> input) {
   if (!input.has_value()) {
     return Id::makeUndefined();
   } else {
-    T res{};
-    auto str = boost::lexical_cast<std::string>(*input);
-    auto conv = std::from_chars(str.data(), str.data() + str.size(), res);
-    if (conv.ec == InvalidArg || conv.ec == OutOfRange) {
+    auto str = absl::StripAsciiWhitespace(input.value());
+    double resD{};
+    auto conv = absl::from_chars(str.data(), str.data() + str.size(), resD);
+    if (conv.ec != std::error_code{} || conv.ptr != str.data() + str.size()) {
       return Id::makeUndefined();
     }
-    if (std::is_same_v<T, double>) { return Id::makeFromDouble(res); }
-    else { return Id::makeFromInt(res); }
+    if constexpr (std::is_same_v<T, int64_t>) { 
+      auto resT = static_cast<T>(resD);
+      return Id::makeFromInt(resT);
+    }
+    else { return Id::makeFromDouble(resD); }
   }
 };
-using MakeStrToInt = StringExpressionImpl<1, decltype(toNumeric<int>)>;
-using MakeDoubleToInt = StringExpressionImpl<1, decltype(toNumeric<double>)>;
+using ToInt = StringExpressionImpl<1, decltype(toNumeric<int64_t>)>;
+using ToDouble = StringExpressionImpl<1, decltype(toNumeric<double>)>;
```

Comment:
This classes then would be called `ToIntExpression` in accordance with the rest.